testdata/p4_16_samples_outputs/psa-register1.p4-error

```diff
@@ -0,0 +1,6 @@
+psa-register1.p4(51): [--Wwarn=unused] warning: reg: unused instance
```

Comment:
why is this warning emitted now? It does not seem unused to me.

Reply:
Because the use get dead-code eliminated (due to being marked as `@noSideEffect`).  Should probably figure out a way to suppress that -- there *was* a use, but it is gone by the time this check occurs.

Reply:
Seems related to #5237.

Reply:
Looking at this specific case, it is coming from `RemoveAllUnusedDeclarations` which calls `RemoveUnusedDeclarations` in a loop.  This pass takes a `warn` flag about whether or not to produce warnings, but when `RemoveAll` is called with `warn=true` it passes that to all invocations of `Remove` in the loop.  It should probably only be passing that in the first iteration, and passing `warn=false` unconditionally in subsequent iterations.  I'll try to figure out an easy way to do that.

Reply:
This seems to be independent of the case is #5237, which is due to ConstantFolding running first.